cypress/component/TableView.cy.tsx

```diff
@@ -0,0 +1,551 @@
+    cy.get(table)
+      .get("thead")
+      .get("tr")
+      .get("th:first-child")
+      .get("label")
+      .get("span")
+      .should("have.attr", "class")
+      .and("to.have.string", "inputContainer");
+    cy.get(table)
+      .get("tbody")
+      .get("tr")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .should("have.attr", "class")
+      .and("to.have.string", "inputContainer");
+  });
+
+  // Should render checkbox inside span
+
+  it("should render checkbox in header", () => {
+    cy.mount(<SelectionTable data-id="table" />);
+    cy.get(table)
+      .get("thead")
+      .get("tr")
+      .get("th:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .should("have.attr", "type", "checkbox")
+      .should("have.attr", "aria-label", "Select All")
+      .should("have.attr", "class")
+      .and("to.have.string", "input");
+    cy.get(table)
+      .get("tbody")
+      .get("tr")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .should("have.attr", "type", "checkbox")
+      .should("have.attr", "aria-label", "Select All")
+      .should("have.attr", "class")
+      .and("to.have.string", "input");
+  });
+
+  // Should select all rows when header checkbox is checked
+
+  it("should select all rows when header checkbox is checked", () => {
+    cy.mount(<SelectionTable data-id="table" />);
+    cy.get(table)
+      .get("thead")
+      .get("tr")
+      .get("th:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .check();
+    cy.get(table)
+      .get("tbody")
+      .get("tr")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .should("be.checked");
+  });
+
+  // Should unselect all rows when header checkbox is unchecked
+
+  it("should unselect all rows when header checkbox is unchecked", () => {
+    cy.mount(<SelectionTable data-id="table" />);
+    cy.get(table)
+      .get("thead")
+      .get("tr")
+      .get("th:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .uncheck();
+    cy.get(table)
+      .get("tbody")
+      .get("tr")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .should("not.be.checked");
+  });
+
+  // Should select a row when checkbox is checked
+
+  it("should select a row when checkbox is checked", () => {
+    cy.mount(<SelectionTable data-id="table" />);
+    cy.get(table)
+      .get("tbody")
+      .get("tr:nth-child(3)")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
+      .check();
+    cy.get(table)
+      .get("tbody")
+      .get("tr:nth-child(3)")
+      .should("have.attr", "aria-selected", "true")
+      .should("have.attr", "class")
+      .and("to.have.string", "isSelected")
+      .get("td:first-child")
+      .get("label")
+      .get("span")
+      .get("input")
```

Comment:
This test currently checks all rows, rather than a single one as per description